server/src/test/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponseTests.java

```diff
@@ -0,0 +1,39 @@
+package org.elasticsearch.action.admin.indices.settings.get;
+
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.RandomCreateIndexGenerator;
+import org.elasticsearch.test.AbstractStreamableXContentTestCase;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+public class GetSettingsResponseTests extends AbstractStreamableXContentTestCase<GetSettingsResponse> {
+    @Override
+    protected GetSettingsResponse createBlankInstance() {
+        return new GetSettingsResponse();
+    }
+
+    @Override
+    protected GetSettingsResponse createTestInstance() {
+        HashMap<String, Settings> indexToSettings = new HashMap<>();
+        int numIndices = randomIntBetween(1, 5);
+        for (int x=0;x<numIndices;x++) {
+            String indexName = randomAlphaOfLength(5);
+            indexToSettings.put(indexName, RandomCreateIndexGenerator.randomIndexSettings());
+        }
+        ImmutableOpenMap<String, Settings> immutableIndexToSettings = ImmutableOpenMap.<String, Settings>builder().putAll(indexToSettings).build();
+        return new GetSettingsResponse(immutableIndexToSettings);
+    }
+
+    @Override
+    protected GetSettingsResponse doParseInstance(XContentParser parser) throws IOException {
+        return GetSettingsResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
would it be possible to rather leave the true default but override `getRandomFieldsExcludeFilter` so that fields are not inserted at the root, but as the same level as `settings` they are, and also not within `settings`? That is because indices and settings names are arbitrary, it doesn't really make sense to insert anything random in there. But at the level of settings, it may make sense. We usually do this to test forward compatibility, to make sure that whatever we may add to the response in the future doesn't break things. In fact I think that your parsing code may be strict to that regard.

Reply:
I attempted to do so -- however it appears that the random fields that get embedded may be fields that the parser is totally unprepared to deal with - for instance, embedding a value object where one is not expected.

Reply:
I suspect that if the test fails it's because your parsing code needs to be adjusted to ignore what it does not know about. We do this in response parsing to ensure forward compatibility.